### PR TITLE
HTCONDOR-1658 Suggested improvements

### PR DESCRIPTION
**Pull request by jasoncpatton**
The diff may be large, but the functionality should be the same.

Most of the changes come from using [the Black code formatter](https://pypi.org/project/black/).
Some other QoL changes made:
1. Used f-strings instead of format statements
2. Made writing to `stderr` easier and directed error messages there.
3. Used [pathlib](https://docs.python.org/3/library/pathlib.html) to turn paths into `Path` objects where it made sense to do so
4. Used [tempfile](https://docs.python.org/3/library/tempfile.html) instead of a hardcoded filename for testing
5. Used [shlex](https://docs.python.org/3/library/shlex.html) splitting to fetch the regexp from the mapfile instead of looking for spaces.

Please check and see that the functionality is still the same after these changes.

### CODE REVIEW

File: src/HTCondor/condor-9-to-10-upgrade-check.py [hunk 1]
```diff
@@ -56,38 +75,42 @@ def getTokenConfiguration(param, is_ce, checking_remote_collector):
...
     cmd = []
     if is_ce:
         cmd.append("condor_ce_config_val")
     else:
         cmd.append("condor_config_val")
     if checking_remote_collector:
         cmd.append("-collector")
-    cmd.extend(["-dump","TRUST_DOMAIN"])
-    p = subprocess.run(cmd,stdout=subprocess.PIPE)
+    cmd.extend(["-dump", "TRUST_DOMAIN"])
```
Comment (jasoncpatton) [inline src/HTCondor/condor-9-to-10-upgrade-check.py:90]: `listA.extend(listB)` is a new one for me, I think you can also do `listA += listB`, but this is fine.

File: src/HTCondor/condor-9-to-10-upgrade-check.py [hunk 2]
```diff
@@ -99,190 +122,250 @@ def getTokenConfiguration(param, is_ce, checking_remote_collector):
...
+    """Read this hosts stored IDTokens and parse out the issuer field"""
+    # Return dict = {issuer:Number tokens issued}
     issuers = {}
-    #Run condor_token_list
-    p = subprocess.run(["condor_token_list"],stdout=subprocess.PIPE)
+    # Run condor_token_list
+    p = subprocess.run(["condor_token_list"], stdout=subprocess.PIPE)
     stored_tokens = p.stdout.rstrip().decode()
-    #Parse each token line for issuer field
+    # Parse each token line for issuer field
```
Comment (jasoncpatton) [inline src/HTCondor/condor-9-to-10-upgrade-check.py:133]: We could probably do this a bit smarter with the `json` library.

File: src/HTCondor/condor-9-to-10-upgrade-check.py (hunk 1 above)
Comment (ColeBollig) [reply src/HTCondor/condor-9-to-10-upgrade-check.py:90]: A little digging and can confirm that listA += listB does also work. However, I will keep this for the time being

File: src/HTCondor/condor-9-to-10-upgrade-check.py (hunk 2 above)
Comment (ColeBollig) [reply src/HTCondor/condor-9-to-10-upgrade-check.py:133]: Good call... I will look into that and attempt to implement

**Review by ColeBollig — APPROVED** [review]
Went through one commit at a time. Overall seems sane. I will take the changes, add a bit more and do some final testing hopefully before V9 EoL next week

**Pull request merged**